internal/cli/cli.go

```diff
@@ -168,27 +161,47 @@ func (c *cli) prepareTenant(ctx context.Context) (Tenant, error) {
 		return Tenant{}, err
 	}
 
-	if t.ClientID != "" && t.ClientSecret != "" {
-		return t, nil
-	}
-
 	if t.AccessToken == "" || scopesChanged(t) {
 		t, err = RunLogin(ctx, c, true)
 		if err != nil {
 			return Tenant{}, err
 		}
-	} else if isExpired(t.ExpiresAt, accessTokenExpThreshold) {
-		// check if the stored access token is expired:
-		// use the refresh token to get a new access token:
+		return t, nil
+	}
+
+	if isExpired(t.ExpiresAt, accessTokenExpThreshold) {
```

Comment:
Much of this diff is due to the elimination of the `else` statement in favor of an early return.